<?xml version="1.0"?>
<launch>

  <arg name="use_test" default="true"/>
  <arg name="use_laser" default="true"/>
  <arg name="use_mapping" default="true"/>
  <!-- Launch the sonar -->
 <!-- <node pkg="laser" type="z_pose" name="z_pose" output="screen"/>-->
  <!-- Launch robot publisher -->
  <node pkg="laser" type="rob_brod.py" name="robot_publisher" output="screen"/>

  <!-- Launch Hokuyo Laser -->
  <group if="$(arg use_laser)">
    <node pkg="hokuyo_node" type="hokuyo_node" name="hokuyo_scans">
      <param name="hokuyo_node/calibrate_time" value="false"/>
      <param name="port" value="/dev/ttyHokuyo"/>
      <param name="frame_id" value="laser_link"/>
      <param name="min_ang" value="-2.2689"/>
      <param name="max_ang" value="2.2689"/>
    </node>
  </group>

  <!-- Launch Hector mapping -->
  <group if="$(arg use_mapping)">
    <include file="$(find laser)/launch/mapping.launch"/>
  </group>

  <!-- Test Launch File --> 
    <group if="$(arg use_test)">
      <include file="$(find laser)/launch/laser_test.launch"/>
    </group> 

</launch>
